probe-rs/src/debug/mod.rs

```diff
@@ -108,10 +109,64 @@ impl From<gimli::ColumnType> for ColumnType {
     }
 }
 
-static CACHE_KEY: AtomicI64 = AtomicI64::new(1);
+/// Object reference as defined in the DAP standard.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ObjectRef {
+    /// Valid object reference (> 0)
+    Valid(NonZeroU32),
+    /// Invalid object reference (<= 0)
+    #[default]
+    Invalid,
+}
+
+impl PartialOrd for ObjectRef {
+    fn partial_cmp(&self, other: &ObjectRef) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for ObjectRef {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        i64::from(*self).cmp(&i64::from(*other))
+    }
+}
+
+impl From<ObjectRef> for i64 {
+    fn from(value: ObjectRef) -> Self {
+        match value {
+            ObjectRef::Valid(v) => v.get() as i64,
+            ObjectRef::Invalid => 0,
+        }
+    }
+}
+
+impl From<i64> for ObjectRef {
+    fn from(value: i64) -> Self {
+        if value < 0 {
+            ObjectRef::Invalid
+        } else {
+            match NonZeroU32::try_from(value as u32) {
+                Ok(v) => ObjectRef::Valid(v),
+                Err(_) => ObjectRef::Invalid,
+            }
+        }
+    }
+}
+
+impl std::str::FromStr for ObjectRef {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let value = s.parse::<i64>()?;
+        Ok(ObjectRef::from(value))
+    }
+}
+
+static CACHE_KEY: AtomicU32 = AtomicU32::new(1);
 /// Generate a unique key that can be used to assign id's to StackFrame and Variable structs.
-pub fn get_sequential_key() -> i64 {
-    CACHE_KEY.fetch_add(1, Ordering::SeqCst)
+pub fn get_object_reference() -> ObjectRef {
```

Comment:
I wonder if we could make this a UUID v4 instead of a global counter.

Reply:
The DAP specifies that the object reference is a int between 1 - 2^31, so an UUID would be too big.

And I think not just using a random number has some advantages. E.g the output order of variables is currently
deterministic because the object references are ordered.

Reply:
That's fair!